include/fmt/color.h

```diff
@@ -492,6 +492,14 @@ template <typename Char> inline void reset_color(buffer<Char>& buffer) {
   buffer.append(reset_color.begin(), reset_color.end());
 }
 
+template <typename Arg> struct styled_arg {
+  FMT_CONSTEXPR styled_arg(const Arg& format_argument, text_style format_style)
+      : argument(format_argument), style(format_style) {}
+
+  const Arg& argument;
```

Comment:
nit: I'd rename `Arg` and `argument` to `T` and `value` respectively because these are just arbitrary types values. What makes them args is the fact that they are passed into a formatting function.